### ISSUE TITLE
Ignore archived subtrees when `jupyter-org-execute-to-point`

### ISSUE THREAD

**Issue by Yevgnen** [dzop/emacs-jupyter#177]
#+property: header-args:jupyter-python :session test :async yes :exports both :tangle test.py

    * Heading1

    #+begin_src jupyter-python
    print('Heading1')
    #+end_src

    * Heading2                                                          :ARCHIVE:

    #+begin_src jupyter-python
    print('Heading2') # Should we ignore this block?
    #+end_src

    * Heading3

    #+begin_src jupyter-python
    print('Heading3') # run `jupyter-org-execute-to-point` here
    #+end_src

When a subtrees is internal archived, shall we ignore any evaluation on it? How do you think about it?

**Comment by dzop** [comment]
Makes sense. I think we could ignore archived subtrees when `point` isn't in an archived subtree, but its not entirely clear what we should do when `point` is within an archived subtree.

If we call `jupyter-org-execute-to-point` when inside, then why shouldn't we call it when outside as well? On the other hand, it seems too restrictive to disallow calling it when inside.

What we do depends on how people use archived subtrees. I personally always put those as the last heading like

```
* Heading1
* Heading2
* Heading 3 :ARCHIVE:
```

whenever I use archives and I don't use `jupyter-org-execute-to-point` too often, so I've never really run into this issue and haven't really formed an opinion on it yet.

I'd like to think about it some more and also invite others to share their opinion on this issue.

**Comment by Yevgnen** [comment]
My scenario is a bit different. I use `jupyter-org-execute-to-point` and `org-babel-execute-buffer` quite often, though the later is not part of `emacs-jupyter`.

I do some calculation or modeling with different methods and structure my org file like

```
* Experiment 
** Preprocessing
** Modeling
*** Method 1           :ARCHIVE:
*** Method 2          :ARCHIVE:
*** Method 3
```

I use the `:ARCHIVE:` tag to disable what methods I've tried. I use these two function the re-evaluate the whole buffer or everything before point just like they don't exist. But I may switch back to any of them (the methods) later.

I know I can set things like `:eval no` to disable evaluation for specific subtree. This approach can not keep the subtree always hidden while archiving does.

So, in my opinion, hidden subtree should always skip. Maybe there's other elegant workflow for me to do this...

**Comment by dzop** [comment]
One way to do what you want is to define a subtree implementing all the pre-processing steps and then use the [noweb-ref](https://orgmode.org/manual/noweb_002dref.html) header argument to merge all those steps into one source block called `preprocess`. Then use `#+CALL: preprocess()` in the `Method` subtrees before the source blocks that implement your method. So in each `Method` subtree you have

```emacs-lisp
* Method 1
#+CALL: preprocess()

#+begin_src jupyter-julia
...
#+end_src
``` 

and then use `org-babel-execute-subtree` (`C-c C-v s`) or use `C-M-<return>` in the Jupyter hydra. 
A full example is below (note I just pushed a change that will get `:results none` to work in the example, 79ddd9357e9250fcb56f658facfbd50130107902).

```emacs-lisp
** Experiment

#+NAME: preprocess
#+HEADER: :var x=1
#+begin_src jupyter-julia :noweb yes :results none
<<preprocess-steps>>
#+end_src
*** Preprocessing
:PROPERTIES:
:header-args: :noweb-ref preprocess-steps
:END:
**** Step 1

#+begin_src jupyter-julia
y = 2*x
#+end_src
*** Modeling
**** Method 1

#+CALL: preprocess(x=10)

#+begin_src jupyter-julia
y
#+end_src

#+RESULTS:
: 20
```
